src/model.js

```diff
@@ -16,19 +17,21 @@ const defaultDesc = process.env.INVOICE_DESC_DEFAULT || 'Lightning Charge Invoic
 
 module.exports = (db, ln) => {
   const newInvoice = async props => {
-    const { currency, amount, expiry, description, metadata, webhook } = props
+    const { currency, amount, expiry, metadata, webhook, lnurlpay_endpoint } = props
 
     const id       = nanoid()
         , msatoshi = props.msatoshi ? ''+props.msatoshi : currency ? await toMsat(currency, amount) : ''
-        , desc     = props.description ? ''+props.description : defaultDesc
-        , lninv    = await ln.invoice(msatoshi || 'any', id, desc, expiry)
+        , desc     = props.descriptionHash || (props.description ? ''+props.description : defaultDesc)
+        , method   = props.descriptionHash ? 'invoicewithdescriptionhash' : 'invoice'
```

Comment:
By the way, let me note here that none of this will work on c-lightning unless you have the method `invoicewithdescriptionhash`, which is provided by [this plugin](https://github.com/fiatjaf/sparko/releases/tag/invoicewithdescriptionhash-v1.1).

Should I check at startup that this method exists by calling `help` and print something to the console if not? Or disable the lnurl-pay thing entirely?

Reply:
I would disable it. The way its implemented makes that pretty easy, the lnurlpay express setup is already self-contained and decoupled from the main app. 

While at it, maybe also make it optional via an `NO_LNURLPAY` option?

I wouldn't bother with disabling the new migrations if lnurlpay is not enabled, just let it create an unused table/column.

As an aside, lnurl-pay could theoretically specify that invoices may use the hash as an hex-encoded string descriptor and that wallets should check for both options, which would make compatibility easier.

Reply:
Awesome, thanks!

Yes, I agree with that last suggestion, but I don't know if it's too late for that by now.

Reply:
What is the rationale for having a `NO_LNURLPAY` option? Users who don't want lnurlpay won't notice anything, the lnurlpay API routes will never be called and the lnurlpay tables will be forever empty and that's it.

On the other hand, if someone wants to use the lnurlpay feature they will very likely miss the fact that there is a `NO_LNURLPAY` option that is on by default and have a lot of frustration.

Reply:
Ok, I did disable the lnurlpay routes when the `invoicewithdescriptionhash` method isn't found, but I think that is also confusing. What if a middleware was defined inside src/lnurl.js that returned a more informative error about the missing plugin (and printed something to the console?) only when lnurlpay routes were called?